### DISCUSSION TITLE
Auto-generated dataframe extension properties for Gradle project

### DISCUSSION THREAD

**Discussion by sripadh8** [Kotlin/dataframe#884] [Q&A]
In order to generate auto-generated extension properties, as mentioned in [https://kotlin.github.io/dataframe/schemasgradle.html#annotation-processing](https://kotlin.github.io/dataframe/schemasgradle.html#annotation-processing), I updated my `brazle.gradle.kts` file to look like this: 

```
plugins {
    kotlin("jvm") version "2.0.20"
    id("org.jetbrains.kotlinx.dataframe") version "0.13.1"
}

dependencies {
    // Kotlin dataframes
    implementation("org.jetbrains.kotlinx:dataframe:0.13.1")

    testImplementation(kotlin("test"))
}
```

As mentioned in the example, I created a sample interface and annotated it with `@DataSchema`. However, I'm not able to access the columns via type-safe accessors. Type-safe accessors are created while working with Kotlin notebooks though, however not in Gradle project. Can someone help me?

**Comment by sripadh8** [comment]
Attaching a sample image to explain further. I executed the project using `./gradlew assemble` as suggested in the documentation. 

<img width="939" alt="Screenshot 2024-09-25 at 1 31 43 PM" src="https://github.com/user-attachments/assets/a07f8aba-6404-4caf-8eaa-32bc735cd29c">

**Comment by Jolanrensen** [comment]
Hi :) just to verify, could you try the same with the just-released version 0.14.0 of DataFrame? It supports kotlin 2.0.20 and KSP, which is used in the generation and is kotlin-version dependent.

**Reply by sripadh8** [reply]
Sure. Will try 0.14.0 and update.

**Reply by sripadh8** [reply]
It works with an upgrade the latest `0.14.0` and repairing the existing IntelliJ caches.

**Comment by koperagen** [comment]
Hi. Apart from verifying the version, there are a few troubleshooting steps 
1. Make sure you run build or more specifically, ksp task. Generated code should appear in the build directory of module where dataframe plugin is used
![image](https://github.com/user-attachments/assets/efb54900-9cf0-4f26-a92f-90ffc8eb409f)
2. If code is there, but it doesn't resolve, maybe it's in another package? Shouldn't be the case since you use it in the same file as interface declaration. Alt+Enter might suggest to import these extensions. If not,
File -> Cache Recovery -> Repair IDE might help. Usually first few steps, up to reopen the project. If it doesn't, there's no need to perform subsequent repair IDE actions after that

**Reply by sripadh8** [reply]
In the morning, I imported ksp and tried as well. All the while `/build/generated/ksp/main/kotlin` directory as empty. I just upgraded to 

```
plugins {
    kotlin("jvm") version "2.0.20"
    id("org.jetbrains.kotlinx.dataframe") version "0.14.0"
}

dependencies {
    // Kotlin dataframes
    implementation("org.jetbrains.kotlinx:dataframe:0.14.0")

    testImplementation(kotlin("test"))
}
```

Then ran the build. At first it doesn't generate still. I then performed File -> Repair IDE, and then ran. It succeeds. It now looks like this: 
<img width="755" alt="Screenshot 2024-09-25 at 4 24 02 PM" src="https://github.com/user-attachments/assets/4614e20a-09ea-4e73-a833-dfbe9c230f39">

**Comment by sripadh8** [comment]
Many thanks for the prompt responses team.
